### DISCUSSION TITLE
Use command line options when launching build from source

### DISCUSSION THREAD

**Discussion by EricLangezaal** [carla-simulator/carla#5470] [General]
For my application I need to continuously change Carla's textures, which is why I am using a build from source. I am noticing that my simulator slows down heavily over time, and eventually crashes with errors that seem related to a lack of GPU/CPU power. 

I am exporting lots of camera data that needs to be realistic, meaning I cannot use no rendering mode or low quality mode. For this reason, I want to try and use [CARLA's offscreen mode](https://carla.readthedocs.io/en/latest/adv_rendering_options/#off-screen-mode). However, since I am using `make launch` to start my simulator, I cannot use its command line option. 

Using `make package` and then launching the .sh file separately does not work either, since I need each new launch to incorporate the new textures and packaging is significantly slower. 

So how do I enable this offscreen mode rendering option in combination with `make launch`?

**Comment by KevinLADLee** [comment]
You could use `make package` command to package your version of carla. Then you could run the carla server like the release version in the `Dist` folder.

**Reply by EricLangezaal** [reply]
As indicated in the post, I need to continuously modify the source code between each run, and due to how much longer packaging takes it is not an option.

**Comment by EricLangezaal** [comment]
I found out that you can specify the command line arguments used when the `make launch` command is executed. This  is done by modifying the command in the "Launch Editor" section of the file `Util/BuildTools/BuildCarlaUE4.sh`
